packages/create-react-app/index.js

```diff
@@ -188,11 +199,34 @@ function run(root, appName, version, verbose, originalDirectory, template) {
     ', and ' + chalk.cyan(packageName) + '...'
   );
   console.log();
-
-  install(allDependencies, verbose, function(code, command, args) {
-    if (code !== 0) {
+  
+  var useYarn = shouldUseYarn();
+  checkIfOnline(useYarn)
+    .then(function(isOnline) {
+      return install(useYarn, allDependencies, verbose, isOnline);
+    })
+    .then(function() {
+      checkNodeVersion(packageName);
+
+      // Since react-scripts has been installed with --save
+      // we need to move it into devDependencies and rewrite package.json
+      // also ensure react dependencies have caret version range
+      fixDependencies(packageName);
+
+      var scriptsPath = path.resolve(
+        process.cwd(),
+        'node_modules',
+        packageName,
+        'scripts',
+        'init.js'
+      );
+      var init = require(scriptsPath);
+      init(root, appName, verbose, originalDirectory, template);
+    })
+    .catch(function() {
       console.log();
       console.error('Aborting installation.', chalk.cyan(command + ' ' + args.join(' ')), 'has failed.');
```

Comment:
I am referring about this commit https://github.com/facebookincubator/create-react-app/pull/1423/commits/c9cece5e67cf8af1b4f208e67766c573aab8df7f